**Linux** [Build 10150](https://ci.wildfly.org/viewLog.html?buildId=23801&buildTypeId=WFPR) outcome was **FAILURE** using a merge of a0cd86e938f6745595ff39140eb0ecc39187a2a3
Summary: Tests failed: 1 (1 new), passed: 3439, ignored: 195 Build time: 01:01:40
### Failed tests
```
org.jboss.as.test.clustering.cluster.web.ConcurrentCoarseWebFailoverTestCase(SYNC-tcp).testGracefulSimpleFailover: java.lang.AssertionError: Session failed to replicate after container 1 was shutdown. expected:<7> but was:<6>
	at org.jboss.as.test.clustering.cluster.web.AbstractWebFailoverTestCase.testFailover(AbstractWebFailoverTestCase.java:211)
	at org.jboss.as.test.clustering.cluster.web.AbstractWebFailoverTestCase.testGracefulSimpleFailover(AbstractWebFailoverTestCase.java:79)
------- Stdout: -------
04:49:06,281 INFO  [org.jboss.as.test.clustering.NodeUtil] (main) Starting deployment=NONE, container=container-0
04:49:06,281 INFO  [org.jboss.as.test.clustering.NodeUtil] (main) Starting deployment=NONE, container=container-1
04:49:06,281 INFO  [org.jboss.as.test.clustering.NodeUtil] (main) Deploying deployment=deployment-0
 [0m04:49:06,296 INFO  [org.jboss.as.repository] (management-handler-thread - 3) WFLYDR0001: Content added at location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/clustering/target/wildfly-SYNC-tcp-0/standalone/data/content/41/28c7f795e8ae0e83b517beb057276e63a3b709/content
 [0m [0m04:49:06,298 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-4) WFLYSRV0027: Starting deployment of "coarse-concurrent-distributable.war" (runtime-name: "coarse-concurrent-distributable.war")
 [0m [33m04:49:06,312 WARN  [org.jboss.as.dependency.private] (MSC service thread 1-8) WFLYSRV0018: Deployment "deployment.coarse-concurrent-distributable.war" is using a private module ("org.jboss.as.clustering.common:main") which may be changed or removed in future versions without notice.
 [0m [33m04:49:06,312 WARN  [org.jboss.as.dependency.private] (MSC service thread 1-8) WFLYSRV0018: Deployment "deployment.coarse-concurrent-distributable.war" is using a private module ("org.jboss.as.server:main") which may be changed or removed in future versions without notice.
 [0m [33m04:49:06,312 WARN  [org.jboss.as.dependency.private] (MSC service thread 1-8) WFLYSRV0018: Deployment "deployment.coarse-concurrent-distributable.war" is using a private module ("org.infinispan:main") which may be changed or removed in future versions without notice.
 [0m [0m04:49:06,316 INFO  [org.jboss.weld.deployer] (MSC service thread 1-8) WFLYWELD0003: Processing weld deployment coarse-concurrent-distributable.war
 [0m [0m04:49:06,319 INFO  [org.jboss.as.ejb3.deployment] (MSC service thread 1-8) WFLYEJB0473: JNDI bindings for session bean named 'TopologyChangeListenerBean' in deployment unit 'deployment "coarse-concurrent-distributable.war"' are as follows:

	java:global/coarse-concurrent-distributable/TopologyChangeListenerBean!org.jboss.as.test.clustering.TopologyChangeListener
	java:app/coarse-concurrent-distributable/TopologyChangeListenerBean!org.jboss.as.test.clustering.TopologyChangeListener
	java:module/TopologyChangeListenerBean!org.jboss.as.test.clustering.TopologyChangeListener
	java:jboss/exported/coarse-concurrent-distributable/TopologyChangeListenerBean!org.jboss.as.test.clustering.TopologyChangeListener
	java:global/coarse-concurrent-distributable/TopologyChangeListenerBean
	java:app/coarse-concurrent-distributable/TopologyChangeListenerBean
	java:module/TopologyChangeListenerBean

 [0m [0m04:49:06,343 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 23) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
 [0m [0m04:49:06,343 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 23) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
 [0m [0m04:49:06,346 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 27) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
 [0m [0m04:49:06,347 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 27) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
 [0m [0m04:49:06,347 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 12) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
 [0m [0m04:49:06,347 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 12) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
 [0m [0m04:49:06,357 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 12) WFLYCLINF0002: Started coarse-concurrent-distributable.war cache from web container
 [0m [0m04:49:06,357 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 27) WFLYCLINF0002: Started routing cache from web container
 [0m [0m04:49:06,366 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 23) WFLYCLINF0002: Started client-mappings cache from ejb container
 [0m [0m04:49:06,527 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 23) WFLYUT0021: Registered web context: /coarse-concurrent-distributable
 [0m [0m04:49:06,546 INFO  [org.jboss.as.server] (management-handler-thread - 3) WFLYSRV0010: Deployed "coarse-concurrent-distributable.war" (runtime-name : "coarse-concurrent-distributable.war")
 [0m04:49:06,556 INFO  [org.jboss.as.test.clustering.NodeUtil] (main) Deploying deployment=deployment-1
 [0m04:49:06,573 INFO  [org.jboss.as.repository] (management-handler-thread - 1) WFLYDR0001: Content added at location /opt/buildAgent/work/a31d203e70e89f90/testsuite/integration/clustering/target/wildfly-SYNC-tcp-1/standalone/data/content/41/28c7f795e8ae0e83b517beb057276e63a3b709/content
 [0m [0m04:49:06,575 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-2) WFLYSRV0027: Starting deployment of "coarse-concurrent-distributable.war" (runtime-name: "coarse-concurrent-distributable.war")
 [0m [33m04:49:06,591 WARN  [org.jboss.as.dependency.private] (MSC service thread 1-3) WFLYSRV0018: Deployment "deployment.coarse-concurrent-distributable.war" is using a private module ("org.jboss.as.clustering.common:main") which may be changed or removed in future versions without notice.
 [0m [33m04:49:06,591 WARN  [org.jboss.as.dependency.private] (MSC service thread 1-3) WFLYSRV0018: Deployment "deployment.coarse-concurrent-distributable.war" is using a private module ("org.jboss.as.server:main") which may be changed or removed in future versions without notice.
 [0m [33m04:49:06,592 WARN  [org.jboss.as.dependency.private] (MSC service thread 1-3) WFLYSRV0018: Deployment "deployment.coarse-concurrent-distributable.war" is using a private module ("org.infinispan:main") which may be changed or removed in future versions without notice.
 [0m [0m04:49:06,598 INFO  [org.jboss.weld.deployer] (MSC service thread 1-3) WFLYWELD0003: Processing weld deployment coarse-concurrent-distributable.war
 [0m [0m04:49:06,601 INFO  [org.jboss.as.ejb3.deployment] (MSC service thread 1-3) WFLYEJB0473: JNDI bindings for session bean named 'TopologyChangeListenerBean' in deployment unit 'deployment "coarse-concurrent-distributable.war"' are as follows:

	java:global/coarse-concurrent-distributable/TopologyChangeListenerBean!org.jboss.as.test.clustering.TopologyChangeListener
	java:app/coarse-concurrent-distributable/TopologyChangeListenerBean!org.jboss.as.test.clustering.TopologyChangeListener
	java:module/TopologyChangeListenerBean!org.jboss.as.test.clustering.TopologyChangeListener
	java:jboss/exported/coarse-concurrent-distributable/TopologyChangeListenerBean!org.jboss.as.test.clustering.TopologyChangeListener
	java:global/coarse-concurrent-distributable/TopologyChangeListenerBean
	java:app/coarse-concurrent-distributable/TopologyChangeListenerBean
	java:module/TopologyChangeListenerBean



```

--- a/jsf/injection/src/main/java/org/jboss/as/jsf/injection/MyFacesInjectionProvider.java
+++ b/jsf/injection/src/main/java/org/jboss/as/jsf/injection/MyFacesInjectionProvider.java
@@ -1,0 +1,70 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.jsf.injection;
+
+import org.apache.myfaces.spi.InjectionProvider;
+import org.apache.myfaces.spi.InjectionProviderException;
+import org.jboss.as.web.common.StartupContext;
+import org.jboss.as.web.common.WebInjectionContainer;
+
+/**
+ * {@link InjectionProvider} implementation which provides MyFaces 2.2 support
+ *
+ * @author Dmitrii Tikhomirov dtikhomi@redhat.com (C) 2016 Red Hat Inc.
+ */
+public class MyFacesInjectionProvider extends InjectionProvider {
+
+    private final WebInjectionContainer injectionContainer;
+
+    public MyFacesInjectionProvider() {
+        this.injectionContainer = StartupContext.getInjectionContainer();
+    }
+
+    @Override
+    public Object inject(Object instance) throws InjectionProviderException {
+        return null;
+    }
+
+    @Override
+    public void postConstruct(Object instance, Object creationMetaData) throws InjectionProviderException {
+        if (injectionContainer != null) {
+            try {
+                injectionContainer.newInstance(instance);
+            } catch (Exception e) {
+                throw new InjectionProviderException(e);
+            }
+        }
+    }
+
+    @Override
+    public void preDestroy(Object instance, Object creationMetaData) throws InjectionProviderException {
+        if (injectionContainer != null) {
+            injectionContainer.destroyInstance(instance);
+        }
+    }
+
+    @Override
+    public boolean isAvailable() {
+        return true;
+    }
+
+}

--- a/jsf/injection/src/main/java/org/jboss/as/jsf/injection/MyFacesInjectionProviderFactory.java
+++ b/jsf/injection/src/main/java/org/jboss/as/jsf/injection/MyFacesInjectionProviderFactory.java
@@ -1,0 +1,53 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2016, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+package org.jboss.as.jsf.injection;
+
+import org.apache.myfaces.spi.InjectionProvider;
+import org.apache.myfaces.spi.InjectionProviderFactory;
+
+import javax.faces.context.ExternalContext;
+
+/**
+ * {@link InjectionProviderFactory} implementation which provides MyFaces 2.2 support
+ *
+ *
+ * @author Dmitrii Tikhomirov dtikhomi@redhat.com (C) 2016 Red Hat Inc.
+ */
+public class MyFacesInjectionProviderFactory extends InjectionProviderFactory {
+
+    private final InjectionProvider injectionProvider;
+
+    public MyFacesInjectionProviderFactory(){
+        injectionProvider = new MyFacesInjectionProvider();
+    }
+
+    @Override
+    public InjectionProvider getInjectionProvider(ExternalContext externalContext) {
+        return injectionProvider;
+    }
+
+    @Override
+    public void release() {
+
+    }
+
+}

--- a/jsf/injection/src/main/resources/META-INF/services/org.apache.myfaces.spi.InjectionProviderFactory
+++ b/jsf/injection/src/main/resources/META-INF/services/org.apache.myfaces.spi.InjectionProviderFactory
@@ -1,0 +1,1 @@
+org.jboss.as.jsf.injection.MyFacesInjectionProviderFactory

--- a/pom.xml
+++ b/pom.xml
@@ -124,7 +124,7 @@
         <version.org.apache.httpcomponents.httpasyncclient>4.1</version.org.apache.httpcomponents.httpasyncclient>
         <version.org.apache.james.apache-mime4j>0.6</version.org.apache.james.apache-mime4j>
         <version.org.apache.lucene>5.3.1</version.org.apache.lucene>
-        <version.org.apache.myfaces.core>2.1.17</version.org.apache.myfaces.core>
+        <version.org.apache.myfaces.core>2.2.10</version.org.apache.myfaces.core>
         <version.org.apache.openjpa>2.4.1</version.org.apache.openjpa>
         <version.org.eclipselink.version>2.6.3</version.org.eclipselink.version>
         <version.org.apache.neethi>3.0.3</version.org.apache.neethi>

